### ISSUE TITLE
UI isn't as pretty as it could be

### ISSUE THREAD

**Issue by cjhammill** [TedDPig123/326_Project#70]
We spent a lot of time making our features actually work, which was kinda necessary for the front end of our website since it relied on the intersection of many features' functionality. However this ended up being a lot of backend work, leaving us little time to make the frontend very beautiful

**Comment by redblue64** [comment]
There were several issues our team had encountered throughout the week, which included scheduling time to work on the project. 

We managed to finalize/push working code of all of our functions on Sunday at around 3pm. Everyone was busy with their other classes, alongside midterms. They have spent the last few weeks making sure their functions were working smoothly. As mentioned in the comment above, most of these functions ended up being back end work and reserved little time for the front end. 

The first issue I encountered was realizing that my sidebar navigation user interface carried few too options to store the many variables present on the home page. I went on youtube, and began to look for more information regarding adding additional features onto my sidebar navigation. I found this video and began to use it as a basis:

https://www.youtube.com/watch?v=NsHhKpR1ItA

After getting half way through reconstructing the sidebar with additional drop downs, I attempted to add element id's the <a> tags. This wasn't possible, as <a> tags are only meant to support hyperlinks. 

I then tried to replace the entire <a> tag with each button instead. Each button that was placed caused the sidebar itself to deform and lose its shape, since the attributes that were tied to the <a> tag were being lose. I then attempted to move the class attributes I placed on the <a> tag onto the buttons, but some attributes weren't compatible and didn't do much visually. 

I opted to scrap the UI I've been working on to try and create a static menu. I went onto w3schools to look for more information. This was one of the websites I used for reference.

https://www.w3schools.com/howto/howto_js_dropdown_sidenav.asp

I quickly discarded the idea after failing to 'push' the site to fit in the side navigation (since it was static, it needed space and therefore shouldn't overlap some features. Knowing I was short on time, I chose to just restructure the buttons already present on the website and give them different background color and a different font. 

While working on this new menu, I contacted the rest of my groupmates and I explained to them the situation. Trying to understand everyone's code 4 hours before the deadline was not the best way to go, so speaking with my groupmates (those who understood their code best) was the best strategy. There were countless times where I'd work on one section of the UI, only for that code to not work due to the properties not coinciding. It didn't help that I had to pay attention where and what I modified in the css file, as everyone's own custom code is in there. I couldn't just change all the properties of a <p> or a <button> tag, as not everything would change due to class names. We all worked on our own UI's separately, but even then there wasn't enough time to fully complete the look of the actual user interface we were hoping for. 

The bottom line is that we waited too long to start properly implementing the user interface, and instead focused more on goals from the next milestone compared to this milestone. My goals for the upcoming milestone is to finish the user interface.
